FFMT-3476 - Packing slip and Email changes

kibo-snehanerekar opened the pull request:
Email template changes below -
item cancel - done
return authorized - done
return request reject - done
return item received - done
return has been updated  - Done
shipment-confirmation.hypr - done
order-cancel.hypr - done
order-pickup-ready.hypr - done 
order-pickup-reminder.hypr -done
order-shipped.hypr - done
partial-pickup-ready.hypr - done
transfer-shipment-created-fulfiller.hypr - done
transfer-shipment-created.hypr - done
transfer-shipment-shipped.hypr - done

Added Packing slip changes

kibo-snehanerekar committed:
FFMT-3476 - Packing slip and Email changes

kibo-snehanerekar committed:
FFMT-3476 - Packing slip and Email changes

kibo-kellytaylor commented on templates/email/order-cancel.hypr:
```diff
@@ -49,6 +49,8 @@
             </thead>
             {% for item in model.items %}
                 <tbody class="mz-ordersummary-lineitems">
+                {% if ((not item.originalLineId) && model.substitutedItems.length > 0) %}
```
@kibo-snehanerekar I see the following conditional statement across most of the templates in your PR and I'm wondering if the check should include an OR for cases where there are no substitutions on the shipment. For example, OR `(not model.substitutedItems.length > 0)` ...maybe I'm reading it wrong.
```
{% if ((not item.originalLineId) && model.substitutedItems.length > 0) %}
```

kibo-kellytaylor reviewed (commented):
@kibo-snehanerekar I see the following conditional statement across the templates in your PR and I'm wondering if the check should include an OR for cases where there are no substitutions on the shipment. 
```
{% if ((not item.originalLineId) && model.substitutedItems.length > 0) %}
```
For example, OR `((not model.substitutedItems)` OR `(not model.substitutedItems.length > 0))` ...maybe I'm reading it wrong.

cc: @Kibo-Omkar

kibo-snehanerekar committed:
FFMT-3476 - Packing slip and Email changes Review changes

kibo-snehanerekar commented:
> 

@kibo-kellytaylor Per discussion with @Kibo-Omkar we no longer required a second condition. The first condition will suffice the requirement to view only Original items hence removing the second condition.

kibo-snehanerekar committed:
FFMT-3476 - Packing slip and Email changes

Label changes per discussion with Amit Solanki

Kibo-Omkar commented on templates/back-office/packing-slip-content.hypr:
```diff
@@ -100,67 +100,75 @@
-                </table>
-                <table class="grid substitute-table">
-                    <thead>
+                {% if model.substitutedItems.length > 0 %}
```
` {% if model.substitutedItems.length > 0 && item.originalLineId %}`

Kibo-Omkar commented on templates/back-office/packing-slip-content.hypr:
```diff
@@ -100,67 +100,75 @@
+                            </thead>
+                        <tbody>
+                            {% for substitutedItem in model.items %}
+                                {% if substitutedItem.originalLineId && substitutedItem.length > 0 %}
```
Can we rename the substitute item and substituted Item (Or Original Items here)?
So that we can easily identify the table grids?
For example 
`{% if substitutedItem.originalLineId && substitutedItems.length > 0 %}`

kibo-snehanerekar replied:
I tried it earlier but sometimes it doesn't work in hypr for some reason hence added seperately

Kibo-Omkar commented on templates/back-office/packing-slip-content.hypr:
```diff
@@ -100,67 +100,75 @@
+                            </tr>
+                            </thead>
+                        <tbody>
+                            {% for substitutedItem in model.items %}
```
As we are showing substitute items, so can we rename the variable accordingly?
` for substituteItem in model.items`

**SubstituteItem** --> _Items are selected as substitutes for an item._
**SubstitutedItem** --> _Items that are substituted with other items._

Kibo-Omkar commented on templates/email/item-cancel.hypr:
```diff
@@ -100,7 +100,52 @@
                 {% endfor %}
-            </table>
+            </table>            
+            {% if model.substitutedItems.length > 0 %}
```
Please check the naming convention while iterating on items.
SubstituteItem --> Items are selected as substitutes for an item.
SubstitutedItem --> Items that are substituted with other items.

kibo-snehanerekar replied:
Changed the naming convention for SubstituteItem

Kibo-Omkar commented on templates/email/order-cancel.hypr:
```diff
@@ -63,11 +65,60 @@
+                    </tr>
+                    </thead>
+                <tbody>
+                    {% for substitutedItem in model.items %}
```
Please check the naming convention while iterating on items.
SubstituteItem --> Items are selected as substitutes for an item.
SubstitutedItem --> Items that are substituted with other items.

Kibo-Omkar commented on templates/email/order-pickup-ready.hypr:
```diff
@@ -224,18 +224,68 @@
+        {% for item in model.items %}
+            {% if (not item.originalLineId) %}
+                {% if (item.quantity > 0) %}
+                    <tr class="mz-ordersummary-line mz-ordersummary-line-item">
```
Can we combine both if statements?
` {% if (not item.originalLineId) &&  (item.quantity > 0) %}`

Kibo-Omkar commented on templates/email/order-pickup-ready.hypr:
```diff
@@ -224,18 +224,68 @@
+                </tr>
+                </thead>
+            <tbody>
+                {% for substitutedItem in model.items %}
```
Please check the comment regarding the naming convention.

Kibo-Omkar commented on templates/email/order-pickup-reminder.hypr:
```diff
@@ -215,9 +217,55 @@
+                </thead>
+            <tbody>
+                {% for substitutedItem in model.items %}
+                    {% if substitutedItem.originalLineId && substitutedItem.length > 0 %}
```
On line 241, we are iterating on  model.items and using "substitutedItem" as a variable (Iterator variable) to access the values from  "the model.items"
On line 242 we are  "substitutedItem.length".
It seems you are checking the "length" property inside the "item" model.

kibo-snehanerekar replied:
changed it to quantity

Kibo-Omkar commented on templates/email/order-shipped.hypr:
```diff
@@ -34,25 +34,75 @@
-                    </td>
-                </tr>
+            {% if ((not packageItem.originalLineId) && package.substitutedItems.length > 0) %}
+                {% if (packageItem.quantity > 0) %}
```
Please check the above comment regarding the "originalLineId"

Kibo-Omkar reviewed (changes requested):
Hi @kibo-snehanerekar ,
As discussed, kindly address the changes.

kibo-snehanerekar committed:
FFMT-3476 - Packing slip and Email changes Review changes

Changed the naming convention to substituteItem , added condition to hide original item header if all quantity gets substituted, Added split packing slip logic, done same changes in email templates where ever required

kibo-snehanerekar commented:
Done with the changes required, except one of adding if condition in one line since I found it breaking at some point

Kibo-Omkar commented on templates/email/order-pickup-ready.hypr:
```diff
@@ -224,18 +232,68 @@
 		{% endfor %}
     </table>
+    
+    {% if model.substitutedItems.length > 0 %}
```
It's been a good practice to check the object before accessing its properties.
`{% if model.substitutedItems && model.substitutedItems.length > 0 %}`

Kibo-Omkar reviewed (approved):
Hi @kibo-snehanerekar ,  It seems there are a few comments that need to be addressed and I am also having a few questions related to that but by considering the priority of the ticket and as per your comment (https://kibo.atlassian.net/browse/FFMT-3476?focusedCommentId=456447)  I have approved the same.
cc: @kibo-kellytaylor 
Thank you.
